### PR TITLE
Expose DNS error codes through the UnknownHostException

### PR DESCRIPTION
**Pull request by tkountis**
Motivation:

UnknownHostException are covering a broader range of discovery failures, where the API consumer has no visibility of the underlying reason. 

Modification:

Expose initial-cause for UnknownHostExceptions to enrich the result. 

Result:

The consumer now can decide whether the exception is due to failures or due an authoritative NXDOMAIN.

### CODE REVIEW

File: resolver-dns/src/main/java/io/netty/resolver/dns/DnsResolveContext.java [hunk 1]
```diff
@@ -654,7 +660,7 @@ private void onResponse(final DnsServerAddressStream nameServerAddrStream, final
                 //                3               Name Error - Meaningful only for
                 //                                responses from an authoritative name
                 //                                server, this code signifies that the
-                //                                domain name referenced in the query does
+                //                                domain name referenced in the query does`
```
Comment (Scottmitch) [inline resolver-dns/src/main/java/io/netty/resolver/dns/DnsResolveContext.java:663]: nit: undo this?

File: resolver-dns/src/main/java/io/netty/resolver/dns/DnsErrorCauseException.java [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    public Throwable fillInStackTrace() {
+        return this;
+    }
+
+    /**
+     * Returns the DNS error-code that caused the {@link UnknownHostException}.
+     *
+     * @return the DNS error-code that caused the {@link UnknownHostException}.
+     */
+    public DnsResponseCode getCode() {
```
Comment (normanmaurer) [inline resolver-dns/src/main/java/io/netty/resolver/dns/DnsErrorCauseException.java:61]: Usually we dont use `get` prefixes... 
```suggestion
    public DnsResponseCode code() {
```

**Review by normanmaurer — CHANGES_REQUESTED** [review]

**Review by bryce-anderson — APPROVED** [review]

**Review by normanmaurer — APPROVED** [review]

**Comment by normanmaurer** [comment]
@tkountis I did retry this build a few times now and it always fails during dns tests… Can you have a look if this might break something ?

**Comment by normanmaurer** [comment]
@tkountis thanks a lot!

**Pull request merged**